src/reference/asciidoc/kafka.adoc

```diff
@@ -563,6 +502,110 @@ CAUTION: If you wish to use the idle event to stop the lister container, you sho
 Instead, you should hand off the event to a different thread that can then stop the container.
 Also, you should not `stop()` the container instance in the event if it is a child container, you should stop the concurrent container instead.
 
+===== Topic/Partition Initial Offset
+
+There are several ways to set the initial offset for a partition.
+
+When manually assigning partitions, simply set the initial offset (if desired) in the configured `TopicPartitionInitialOffset` arguments (see <<message-listener-container>>).
+You can also seek to a specific offset at any time.
+
+When using group management where the broker assigns partitions:
+
+- For a new `group.id`, the initial offset is determined by the `auto.offset.reset` consumer property (`earliest` or `latest`).
+- For an existing group id, the initial offset is the current offset for that group id.
+You can, however, seek to a specific offset during initialization (or at any time thereafter).
+
+===== Seeking to a Specific Offset
+
+In order to seek, your listener must implement `ConsumerSeekAware` which has the following methods:
+
+[source, java]
+----
+void registerSeekCallback(ConsumerSeekCallback callback);
+
+void onPartionsAssigned(Collection<TopicPartitionCurrentOffset> assignments, ConsumerSeekCallback callback);
+----
+
+The first is called during initialization; this callback should be used when seeking at some arbitrary time after initialization.
+You should save a reference to the callback; if you are using the same listener in multiple containers (or in a `ConcurrentMessageListenerContainer`) you should store the callback in a `ThreadLocal` or some other structure keyed by the listener `Thread`.
```

Comment:
He-he!
> keyed by the listener `Thread`.

That leads only to the `ThreadLocal` logic:
```
 /* ThreadLocal values pertaining to this thread. This map is maintained
  * by the ThreadLocal class. */
ThreadLocal.ThreadLocalMap threadLocals = null;
```
I don't see reason to do some other storage based on `Thread` key, but, eh, let it be, it is end-user choice.